testdata/p4tc_samples_outputs/simple_extern_example_control_blocks.c

```diff
@@ -58,8 +58,8 @@ static __always_inline int process(struct __sk_buff *skb, struct my_ingress_head
     hdr = &(hdrMd->cpumap_hdr);
     meta = &(hdrMd->cpumap_usermeta);
 {
-        struct p4tc_ext_bpf_params* ext_params;
-        struct p4tc_ext_bpf_val* ext_val;
+        struct p4tc_ext_bpf_params* ext_params = malloc(sizeof(struct p4tc_ext_bpf_params));
+        struct p4tc_ext_bpf_val* ext_val = malloc(sizeof(struct p4tc_ext_bpf_val));
```

Comment:
The issue here is that `malloc` is not supported by ebpf
You'd have to use something like a ring buffer map (https://www.kernel.org/doc/html/next/bpf/ringbuf.html) to make this work
I think it was complex for the compiler to emit this as stack variable, but working around dynamic allocation in ebpf is also not that simple

Reply:
switched to structure instead of structure pointer.